### PR TITLE
Setting up Ansible Linter

### PR DESCRIPTION
**Pull request by Harshil-Jani**
This PR closes https://github.com/OpenTermsArchive/engine/issues/949 where it was mentioned about setting up Ansible Linter. Currently, I have set this up in my fork and it throws some errors as expected.

So, Here I would love to incorporate the feedback by the community whether you want to disable those errors/warnings or fix the files in order to pass the CI. 
Here is the failed CI run : https://github.com/Harshil-Jani/deployment/actions/runs/4444762373/jobs/7803258443#step:4:107

### CODE REVIEW

**Comment by Ndpnt** [comment]
Hi @Harshil-Jani, thanks for your contribution! 👍 

I checked the errors reported by `ansible-lint` and here are my suggestions:
- `command-instead-of-module`: should be fixed.
- `yaml[line-length]`: should be fixed.
- `name[template]`: should be ignored. 
- `package-latest`: should be ignored. For `nodejs` and `mongo` major version is ensured by defining on our side the source from which the package will be installed. For all common required packages and for chromium it is acceptable to always install the latest version.
- `no-changed-when`: should be fixed.
- `galaxy[version-incorrect]`: should be ignored. Currently, having a collection version under `1.0.0` it is voluntary because our [public API is not really defined and is not stable](https://semver.org/#spec-item-5) and for now we do not [worry a lot about backwards compatibility](https://semver.org/#how-do-i-know-when-to-release-100).

If you are still interested in incorporating this feedback, it would be a pleasure for us, otherwise we will take care of it, just let us know :wink:.

**Comment by Harshil-Jani** [comment]
Hey !! I need some help for `no-change-when` and `command-instead-of-module`. I am not able to fix them up. Can someone exactly point at how this should be done ?

**Comment by Ndpnt** [comment]
#### For `no-change-when` rule

The idea is to tell Ansible if there really has been a change on tasks that do not naturally detect if a change has occurred or not. 
To do this, you can use a combination of `register` and `changed_when`. You record the output of the task with `register: <var_name>` and you define a condition to define when a particular task has changed with `changed_when`.

- For example, for the error in `roles/engine/tasks/database.yml` line 37. The task fetch origin repository.
If the `git fetch origin` command returns nothing, it means that there is nothing to fetch.
So the error, can be fixed in `roles/engine/tasks/database.yml` line 37 with:
```
- name: Get latest data from {{ repository }}
  ansible.builtin.command:
    cmd: git fetch origin
    chdir: '{{ directory }}'
  register: fetch
  changed_when: fetch.msg != ''
  tags:
    - restart
    - start
    - update
    - setup
```

- Another example for the error in `roles/engine/tasks/main` line 15. The task installs node dependencies in production.

The `npm install --production` command returns something like that when it has installed some packages:
```
…
added 847 packages, and audited 848 packages in 8s
…
```

or something like that if packages were already been installed:
```
…
up to date, audited 848 packages in 1s
…
```
So the error can be fixed in `roles/engine/tasks/main` line 15 with:
```
- name: Install services declarations dependencies
  ansible.builtin.command:
    cmd: npm install --production
    chdir: '/home/{{ ansible_user }}/{{ ota_declarations_directory }}'
  register: npm_install
  changed_when: npm_install.msg | regex_search("added [0-9]+ packages")
  tags:
    - setup
    - update
    - update-declarations
```

#### For `command-instead-of-module` rule

The idea is to use a specific Ansible module instead of using directly a command. The errors we have here are related to `git` and it is suggested to use the native [ansible git module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/git_module.html). I think it's not always possible. For example I do not think we can do a hard reset with the native git module. I you succeed to use the native git module it's better, otherwise you can add this type of error in the warning list.

**Comment by Harshil-Jani** [comment]
In the ansible.builtin.git documentation I cannot figure out a way to git reset hard. As of now, I am adding the `command-instead-of-module` rule in warning list.

**Comment by Harshil-Jani** [comment]
Whew !! Finally, CI lints passes. Let me know, If there is any alternative to the warning modules which we can leverage to make the linting more robust. Else this should be ready to be merged.

File: .config/ansible-lint.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+warn_list:
```
Comment (Ndpnt) [inline .config/ansible-lint.yml:1]: Can you move the content of this file from `.config/ansible-lint.yml` to `.ansible-lint`?

File: .github/workflows/ansible-lint.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: ansible-lint
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Ansible Lint # Naming the build is important to use it as a status check
```
Comment (Ndpnt) [inline .github/workflows/ansible-lint.yml:6]: Can you remove all the comments imported from the [ansible-lint-action](https://github.com/ansible/ansible-lint-action) example from in this file?

File: roles/engine/tasks/database.yml [hunk 3]
```diff
@@ -7,7 +7,9 @@
     update: false
     accept_hostkey: true
     key_file: '/home/{{ ansible_user }}/.ssh/ota-bot-key'
-  register: existing_repository # the `before` property of the return value can tell us if the repository has been cloned already or not, see <https://docs.ansible.com/ansible/latest/collections/ansible/builtin/git_module.html#return-values>
+  register: existing_repository
```
Comment (Ndpnt) [inline roles/engine/tasks/database.yml:10]: It's better to have block comments above the related code. can you move following comments before `register: existing_repository`?

**Review by Ndpnt — CHANGES_REQUESTED** [review]
Hi @Harshil-Jani,

Currently all `changed_when` statements don't work properly because `msg` seems to be almost always empty. 
Fixing them is more complicated than that and depends on the situation. 
I tried to fix them quickly but it's not that easy, and it involves some major changes that are outside the scope of this PR. Using external roles to install NodeJS and MongoDB for example or adding tasks in other cases, and even rethinking some tasks that are currently not very well designed.

So I suggest to remove all changes introduced in your commit `595db34: fix : no-changed-when` and add `no-changed-when` in the warning list for now.

File: .config/ansible-lint.yml (hunk 1 above)
Comment (Harshil-Jani) [reply .config/ansible-lint.yml:1]: This is the default location for the config file. The CI action which we are using is picking config from `.config` directory itself.

Comment (Harshil-Jani) [reply .config/ansible-lint.yml:1]: As an alternative, We would want to tweak entire action code and use it directly here.

Comment (Ndpnt) [reply .config/ansible-lint.yml:1]: I saw that you named the file `ansible-lint.yml` but to be recognised by Ansible lint, the file should not have an extension and it should begin with a `.` like this `.ansible-lint`. Can you try to rename the file?

**Comment by Ndpnt** [comment]
We are almost done :), as I can't push to your branch, can you add the following entry in the changelog?

```
## Unreleased [minor]

### Added
- Automate linting with GitHub Actions
```

_Do not forget to add `[minor]` after the title `## Unreleased`_

**Comment by Ndpnt** [comment]
Thank you very much @Harshil-Jani for your contribution to making this Ansible collection more robust. 👍

**Comment by Harshil-Jani** [comment]
Thankyou for reviewing, helping and merging. ✌🏻

**Pull request merged**